### PR TITLE
I accidentally [deliberately] killed it [the ComparisonTestFramework]

### PR DESCRIPTION
**Pull request by jnewbery**
Builds on #11772, #11773 and #11817. Please review those PRs first.

Final step in #10603.

- First commit removes bip9-softforks.py.  bip9-sofforks.py was intended to be a generic test for versionbits deployments. However, it only tests CSV activation and was not updated to test segwit activation. CSV activation is tested by bip68-112-113-p2p.py, so this test is duplicated effort. Rather than try to update it to use the BitcoinTestFramework, just remove it. (see https://github.com/btcdrak/bitcoin/pull/8 for previous discussion around the redundancy of bip9-softforks.py)
- Second commit removes the now unused BitcoinComparisonFramework class and the comptool and blockstore modules.

### CODE REVIEW

**Review by ryanofsky — COMMENTED** [review]
Reviewed just the two removal commits, assuming all the other commits will go away when this is rebased.

- utACK 5e483e1eae34083ebfe7bc89d21644fa2c8f4389 [tests] Remove Comparison Test Framework
- utACK 8be729761f794d92b5bec2262ebee9fea35e518a [tests] Remove bip9-softforks.py

**Comment by jnewbery** [comment]
Rebased to remove merged commits.

**Comment by jnewbery** [comment]
Rebased. This should be a very easy review, since it's just removing dead code.

**Comment by sdaftuar** [comment]
utACK!  Thanks for sticking with this.

File: test/functional/README.md [hunk 1]
```diff
@@ -89,54 +89,6 @@ thread.)
...
-  - `sync_every_transaction`: `True/False`.  Analogous to
-    `sync_every_block`, except if the outcome on the last tx is "None",
-    then the contents of the entire mempool are compared across all bitcoind
-    connections.  If `True` or `False`, then only the last tx's
-    acceptance is tested against the given outcome.
-
-- For examples of tests written in this framework, see
-  `p2p_invalid_block.py` and `feature_block.py`.
-
-### test-framework modules
```
Comment (MarcoFalke) [inline test/functional/README.md]: nit: Is this line supposed to go away?

Comment (jnewbery) [reply test/functional/README.md]: Good spot! Added that back, and removed the documentation for `test_framework/blockstore.py`

**Comment by MarcoFalke** [comment]
utACK 9c92c8c82716d0c35b638142a1125d07b3f240a8

**Pull request merged**